pkg/controllers/corecluster/corecluster_controller.go

```diff
@@ -215,6 +219,52 @@ func (r *CoreClusterReconciler) ensureCoreClusterControlPlaneInitializedConditio
 	return nil
 }
 
+// setControllerConditionsToNormal sets the CoreClusterController conditions to the normal state.
+func (r *CoreClusterController) setControllerConditionsToNormal(ctx context.Context, log logr.Logger) error {
+	co, err := r.GetOrCreateClusterOperator(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster operator: %w", err)
+	}
+
+	conds := []configv1.ClusterOperatorStatusCondition{
+		operatorstatus.NewClusterOperatorStatusCondition(operatorstatus.CoreClusterControllerAvailableCondition, configv1.ConditionTrue, operatorstatus.ReasonAsExpected,
+			"Core Cluster Controller works as expected"),
+		operatorstatus.NewClusterOperatorStatusCondition(operatorstatus.CoreClusterControllerDegradedCondition, configv1.ConditionFalse, operatorstatus.ReasonAsExpected,
+			"Core Cluster Controller works as expected"),
+	}
+
+	log.V(2).Info("Core Cluster Controller is Available")
+
+	if err := r.SyncStatus(ctx, co, conds); err != nil {
+		return fmt.Errorf("failed to sync cluster operator status: %w", err)
+	}
+
+	return nil
+}
+
+// setControllerConditionDegraded sets the CoreClusterController conditions to a degraded state.
+func (r *CoreClusterController) setControllerConditionDegraded(ctx context.Context, log logr.Logger, reconcileErr error) error {
+	co, err := r.GetOrCreateClusterOperator(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster operator: %w", err)
+	}
+
+	conds := []configv1.ClusterOperatorStatusCondition{
+		operatorstatus.NewClusterOperatorStatusCondition(operatorstatus.CoreClusterControllerAvailableCondition, configv1.ConditionTrue, operatorstatus.ReasonAsExpected,
+			"Core Cluster Controller works as expected"),
+		operatorstatus.NewClusterOperatorStatusCondition(operatorstatus.CoreClusterControllerDegradedCondition, configv1.ConditionTrue, operatorstatus.ReasonAsExpected,
+			fmt.Sprintf("Core Cluster Controller is degraded: %s", reconcileErr.Error())),
```

Comment:
If on the first ever reconcile in a cluster, we have an error, and set ourselves degraded, should this really be Available=True?

I'm wondering if managing the available condition separately would be a better way to handle this?